File: Giffer/Models/PlaybackMode.swift
import Foundation

enum PlaybackMode: String, CaseIterable, Identifiable {
    case forward = "Forward"
    case reverse = "Reverse"
    case bounce = "Bounce"

    var id: String { rawValue }

    var icon: String {
        switch self {
        case .forward: return "play"
        case .reverse: return "backward"
        case .bounce:  return "repeat"
        }
    }
}

File: android/app/src/main/java/com/leaptools/giffer/model/PlaybackMode.kt
package com.leaptools.giffer.model

import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.PlayArrow
import androidx.compose.material.icons.filled.Repeat
import androidx.compose.material.icons.automirrored.filled.Undo
import androidx.compose.ui.graphics.vector.ImageVector

enum class PlaybackMode(val label: String, val icon: ImageVector) {
    FORWARD("Forward", Icons.Filled.PlayArrow),
    REVERSE("Reverse", Icons.AutoMirrored.Filled.Undo),
    BOUNCE("Bounce", Icons.Filled.Repeat);
}
